src/virtualbox-template.xml

```diff
@@ -0,0 +1,127 @@
+      </Item>
+      <Item>
+        <rasd:Address>0</rasd:Address>
+        <rasd:Caption>sataController0</rasd:Caption>
+        <rasd:Description>SATA Controller</rasd:Description>
+        <rasd:ElementName>sataController0</rasd:ElementName>
+        <rasd:InstanceID>3</rasd:InstanceID>
+        <rasd:ResourceSubType>AHCI</rasd:ResourceSubType>
+        <rasd:ResourceType>20</rasd:ResourceType>
+      </Item>
+     <Item>
+        <rasd:Address>0</rasd:Address>
+        <rasd:Caption>usb</rasd:Caption>
+        <rasd:Description>USB Controller</rasd:Description>
+        <rasd:ElementName>usb</rasd:ElementName>
+        <rasd:InstanceID>4</rasd:InstanceID>
+        <rasd:ResourceType>23</rasd:ResourceType>
+      </Item>
+      <Item>
+        <rasd:AddressOnParent>0</rasd:AddressOnParent>
+        <rasd:Caption>disk1</rasd:Caption>
+        <rasd:Description>Disk Image</rasd:Description>
+        <rasd:ElementName>disk1</rasd:ElementName>
+        <rasd:HostResource>/disk/vmdisk1</rasd:HostResource>
+        <rasd:InstanceID>5</rasd:InstanceID>
+        <rasd:Parent>3</rasd:Parent>
+        <rasd:ResourceType>17</rasd:ResourceType>
+      </Item>
+      <Item>
+        <rasd:AutomaticAllocation>true</rasd:AutomaticAllocation>
+        <rasd:Caption>Ethernet adapter on 'NAT'</rasd:Caption>
+        <rasd:Connection>NAT</rasd:Connection>
+        <rasd:ElementName>Ethernet adapter on 'NAT'</rasd:ElementName>
+        <rasd:InstanceID>6</rasd:InstanceID>
+        <rasd:ResourceSubType>E1000</rasd:ResourceSubType>
+        <rasd:ResourceType>10</rasd:ResourceType>
+      </Item>
+     </VirtualHardwareSection>
+     <vbox:Machine ovf:required="false" version="1.16-linux" uuid="fdee0598-5b31-4597-a441-36575dda937a" name="{vsphere_image_name}" OSType="Fedora_64" >
+      <ovf:Info>Complete VirtualBox machine configuration in VirtualBox format</ovf:Info>
+      <Hardware>
+        <CPU count="{ovf_cpu_count}">
+          <PAE enabled="true"/>
+          <LongMode enabled="true"/>
+          <X2APIC enabled="true"/>
+          <HardwareVirtExLargePages enabled="false"/>
+        </CPU>
+        <Memory RAMSize="{ovf_memory_mb}"/>
+        <Firmware type="EFI"/>
+        <Display controller="VMSVGA" VRAMSize="36"/>
+        <BIOS>
+          <IOAPIC enabled="true"/>
+          <SmbiosUuidLittleEndian enabled="true"/>
+        </BIOS>
+        <USB>
+          <Controllers>
+            <Controller name="OHCI" type="OHCI"/>
+          </Controllers>
+        </USB>
+        <Network>
+          <Adapter slot="0" enabled="true" type="virtio">
+            <NAT/>
+          </Adapter>
+        </Network>
+        <Clipboard mode="Disabled"/>
+      </Hardware>
+      <StorageControllers>
+        <StorageController name="AHCI" type="AHCI" PortCount="16" useHostIOCache="false" Bootable="true" IDE0MasterEmulationPort="0" IDE0SlaveEmulationPort="1" IDE1MasterEmulationPort="2" IDE1SlaveEmulationPort="3">
```

Comment:
I noticed this in the kernel logs when booting a test instance:

```
[    1.004327] ata2: SATA link down (SStatus 0 SControl 300)
[...]
[    1.234590] RAS: Correctable Errors collector initialized.
[    1.736404] ata3: SATA link down (SStatus 0 SControl 300)
[    2.260161] ata4: SATA link down (SStatus 0 SControl 300)
[    2.783381] ata5: SATA link down (SStatus 0 SControl 300)
[    3.234328] ata6: SATA link down (SStatus 0 SControl 300)
[    4.063800] ata7: SATA link down (SStatus 0 SControl 300)
[    4.832404] ata8: SATA link down (SStatus 0 SControl 300)
[    5.280195] ata9: SATA link down (SStatus 0 SControl 300)
[    5.847983] ata10: SATA link down (SStatus 0 SControl 300)
[    6.302791] ata11: SATA link down (SStatus 0 SControl 300)
[    7.077747] ata12: SATA link down (SStatus 0 SControl 300)
[    7.912488] ata13: SATA link down (SStatus 0 SControl 300)
[    8.405535] ata14: SATA link down (SStatus 0 SControl 300)
[    9.084071] ata15: SATA link down (SStatus 0 SControl 300)
[    9.516715] ata16: SATA link down (SStatus 0 SControl 300)
[    9.533477] Freeing unused decrypted memory: 2036K
```

Note the timestamps.  So on my system, we're wasting 8.3 seconds during boot waiting for timeouts on each of the unused AHCI ports.  I'm not sure why the timeouts are taking so long, but maybe we should reduce the port count.  I reduced it to 1 and got:

```
[    1.109814] RAS: Correctable Errors collector initialized.
[    1.111956] Freeing unused decrypted memory: 2036K
```

I then tried adding an empty hard disk to the VM with the GUI, and an empty DVD drive with `VBoxManage`:

```
VBoxManage storageattach z --storagectl AHCI --medium emptydrive --type dvddrive --port 2
```

and both changes correctly added a SATA port to the machine when adding the drive, without erroring.  So it seems to me that we can reduce the port count to 1 without causing UX issues if the user wants to add more drives.

Reply:
Ah ok, thats easy let me get that in.